pylint/utils.py

```diff
@@ -704,48 +709,145 @@ def add_renamed_message(self, old_id, old_symbol, new_symbol):
+        self._check_id_and_symbol_consistency(message.msgid, message.symbol)
+        self._check_symbol(message.msgid, message.symbol)
+        self._check_msgid(message.msgid, message.symbol)
+        for old_name in message.old_names:
+            self._check_symbol(message.msgid, old_name[1])
+        self._messages[message.symbol] = message
+        self._register_alternative_name(message, message.msgid, message.symbol)
+        for old_id, old_symbol in message.old_names:
+            self._register_alternative_name(message, old_id, old_symbol)
+        self._msgs_by_category[message.msgid[0]].append(message.msgid)
+
+    @staticmethod
+    def _check_checker_consistency(messages):
+        """ Check the msgid consistency in a list of messages definitions.
+
+        msg ids for a checker should be a string of len 4, where the two first
+        characters are the checker id and the two last the msg id in this
+        checker.
+
+        :param list messages: List of MessageDefinition.
+        :raises InvalidMessageError: If the checker id in the messages are not
+        always the same """
+        checker_id = None
+        existing_ids = []
+        for message in messages:
+            if checker_id is not None and checker_id != message.msgid[1:3]:
+                error_msg = f"Inconsistent checker part in message id "
+                error_msg += f"'{message.msgid}' (expected 'x{checker_id}xx' "
+                error_msg += f"because we already had {existing_ids})."
+                raise InvalidMessageError(error_msg)
+            checker_id = message.msgid[1:3]
+            existing_ids.append(message.msgid)
+
+
+    def _register_alternative_name(self, msg, msgid, symbol):
         """helper for register_message()"""
-        if name in self._messages and self._messages[name] != msg:
-            raise InvalidMessageError(
-                'Message symbol %r is already defined' % name)
-        if name in self._alternative_names and self._alternative_names[name] != msg:
-            raise InvalidMessageError(
-                'Message %s %r is already defined' % (
-                    'id' if len(name) == 5 and name[0] in MSG_TYPES else 'alternate name',
-                    name))
-        self._alternative_names[name] = msg
+        self._check_id_and_symbol_consistency(msgid, symbol)
+        self._alternative_names[msgid] = msg
+        self._alternative_names[symbol] = msg
+
+    def _check_symbol(self, msgid, symbol):
+        """ Check that a symbol is not already used. """
+        if symbol in self._messages:
+            other_msgid = self._messages[symbol].msgid
+            MessagesStore._raise_duplicate_msg_id(symbol, msgid, other_msgid)
+        if symbol in self._alternative_names:
+            msg = self._alternative_names[symbol]
+            if msg.symbol == symbol:
+                other_msgid = msg.msgid
+            else:
+                for old_msgid, old_symbol in msg.old_names:
```

Comment:
Like in a lot of other places I copypasted the existing code and just created function with identical behavior. I'll just add the initialization to None, because this function will probably be deleted with #2036.